### DISCUSSION TITLE
How to update an API's description

### DISCUSSION THREAD

**Discussion by haroldrandom** [fastapi/fastapi#6582] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.get("/")
async def read_items():
    """
    doc: List all items
    """

    logger.info(f"List All items: [{fake_items_db}]")

    return fake_items_db
```

### Description

I want to change the API's description from the default one "Read Items" to my customized text, like "List All items".

How can I do that?

### Operating System

Linux, Windows, macOS

### FastAPI Version

0.63.0

### Python Version

3.8.3

**Comment by STeveShary** [comment, marked as answer]
@haroldrandom You can update the description and many other things by setting parameters in the `@router.get("/")` function.  The `get()` call will call the `add_api_route()` function in the FastAPI code: https://github.com/tiangolo/fastapi/blob/master/fastapi/routing.py#L479.  You can  set any of the following attributes.   You may be especially interested in `summary` and `description`

Example:
```python3
@router.get("/", description="List all items")
async def read_items():
    """
    doc: List all items
    """

    logger.info(f"List All items: [{fake_items_db}]")

    return fake_items_db

```

**Comment by haroldrandom** [comment]
Thanks, @STeveShary, that's a big help!
